models/users.go

```diff
@@ -449,3 +469,51 @@ func (p PasswordReset) BuildCognitoRequest(clientSecret string, clientId string)
 		Username:   &p.Email,
 	}
 }
+
+// BuildListUserGroupsRequest build the require input for cognito query to obtain the groups for given user
+func (p UserParams) BuildListUserGroupsRequest(userPoolId string, nextToken string) *cognitoidentityprovider.AdminListGroupsForUserInput {
+
+	if nextToken != "" {
+		return &cognitoidentityprovider.AdminListGroupsForUserInput{
+			UserPoolId: &userPoolId,
+			Username:   &p.ID,
+			NextToken:  &nextToken,
+		}
+	}
+
+	return &cognitoidentityprovider.AdminListGroupsForUserInput{
+		UserPoolId: &userPoolId,
+		Username:   &p.ID}
+
+}
+
+//BuildListUserGroupsSuccessfulJsonResponse
+// formats the output to comply with current standards and to json , adds the count of groups returned and
+func (p *ListUserGroups) BuildListUserGroupsSuccessfulJsonResponse(ctx context.Context, result *cognitoidentityprovider.AdminListGroupsForUserOutput) ([]byte, error) {
+	for _, tmpGroup := range result.Groups {
```

Comment:
As `result` is a pointer it can be `nil` so we should add a 
```
if result == nil {}
```
Check before we do the loop - if `result` is nil then `result.Groups` will error with nil pointer 

Simplified example: https://play.golang.org/p/Xps38s8iawz